xdsl/dialects/vector.py

```diff
@@ -169,4 +169,74 @@ def get(memref: SSAValue | Operation, indices: List[SSAValue | Operation],
                                 ])
 
 
-Vector = Dialect([Load, Store, Broadcast, FMA, Maskedload], [])
+@irdl_op_definition
+class Maskedstore(Operation):
+    name = "vector.maskedstore"
+    memref: Annotated[Operand, MemRefType]
+    indices: Annotated[VarOperand, IndexType]
+    mask: Annotated[Operand, VectorType]
+    value_to_store: Annotated[Operand, VectorType]
+
+    def verify_(self):
+        memref_element_type = self.memref.typ.element_type
+
+        if memref_element_type != self.value_to_store.typ.element_type:
+            raise VerifyException(
+                "MemRef element type should match the stored vector type.")
```

Comment:
Maybe report the types in this error message?

Reply:
For consistency, I'll create a separate PR to modify type check warnings in a similar fashion for other operations as well once this patch lands (I can do that in this PR itself but I think those changes will be unrelated to the PR/commit title and it might not be a good idea to fuse them here for a clean commit history).